### PR TITLE
test(tile): add accesibility tests

### PR DESCRIPTION
**Pull request by DlgSHi**
#### Button
- Add `accessibility` Cypress tests for the `Tile` and `Tile Select` components to use the new cypress-component-react framework for testing.
- Move the testing component to the `tile-test.stories.tsx` and `tile-select-test.stories.tsx`

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added tests for `accessibility`
- [x] Check if the `tile.test.js` and `tile-select.test.js` files passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Tile` and `Tile Select` tests are not running twice.

### CODE REVIEW

File: cypress/support/accessibility/a11y-utils.js [hunk 1]
```diff
@@ -114,6 +114,7 @@ export default (from, end) => {
       !prepareUrl[0].startsWith("popover-container") &&
       !prepareUrl[0].startsWith("menu") &&
       !prepareUrl[0].startsWith("inline-inputs") &&
+      !prepareUrl[0].startsWith("tile") &&
```
Comment (stephenogorman) [inline cypress/support/accessibility/a11y-utils.js:117]: should this also include:
      !prepareUrl[0].startsWith("tile-select") &&

File: src/components/tile/tile.test.js [hunk 2]
```diff
@@ -255,7 +207,7 @@ context("Tests for Tile component", () => {
       "should check border variant as %s for tile component",
```
Comment (stephenogorman) [inline src/components/tile/tile.test.js:207]: "should check border variant as %s for **Tile** component",

File: src/components/tile/tile.test.js [hunk 3]
```diff
@@ -270,11 +222,171 @@ context("Tests for Tile component", () => {
...
+      "should check Accessibility for w as %s for Tile component",
+      (w) => {
+        CypressMountWithProviders(<testStories.DlTileComponent w={w} />);
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(["default", "selected", "positive", "negative", "caution", "info"])(
+      "should check Accessibility for border variant as %s for tile component",
```
Comment (stephenogorman) [inline src/components/tile/tile.test.js:365]: "should check Accessibility for border variant as %s for **Tile** component",

File: src/components/tile/tile.test.js [hunk 4]
```diff
@@ -270,11 +222,171 @@ context("Tests for Tile component", () => {
...
+    );
+
+    it.each([
+      "borderWidth000",
+      "borderWidth100",
+      "borderWidth200",
+      "borderWidth300",
+      "borderWidth400",
+    ])(
+      "should check Accessibility for border width as %s for tile component",
```
Comment (stephenogorman) [inline src/components/tile/tile.test.js:382]: "should check Accessibility for border width as %s for **Tile** component",

File: cypress/support/accessibility/a11y-utils.js (hunk 1 above)
Comment (DlgSHi) [reply cypress/support/accessibility/a11y-utils.js:117]: no, becase `tile-select` as well starts with `tile` :)

File: src/components/tile/tile.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/tile/tile.test.js]: updated

File: src/components/tile/tile.test.js (hunk 3 above)
Comment (DlgSHi) [reply src/components/tile/tile.test.js]: updated

File: src/components/tile/tile.test.js (hunk 4 above)
Comment (DlgSHi) [reply src/components/tile/tile.test.js]: updated

**Review by stephenogorman — DISMISSED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**